### PR TITLE
Update gem descriptions, README and refactors

### PR DESCRIPTION
**Pull request by bwillis**
This gets it closer to a releasable version by clarifying the interface and updating the documentation. 

## Interface

There are 2 parts that are important, the Logger  and the LogFormatter. The Logger here wraps the fluent-logger service, which is pretty straightfoward and exists elsewhere (acts-as-fluent-logger). The formatter is really the key, you can use the default `ActiveSupport::TaggedLogging` for tagged logging, but it hardcodes the tag formats as prefixed strings, which if you are formatting as JSON, it is not valid JSON. Here, we provide the `FluentLoggerRails::TaggedHashFormatter` which maintains tags in a structure that when passed to fluent-logger will format as valid JSON. We used to have JSON formatter explicitly, but this is really an output format controlled by fluentd.

## Documentation

I updated the Gemfile description and the README.

### CODE REVIEW

File: spec/integration/fluent_logger_rails/logger_spec.rb [hunk 1]
```diff
@@ -20,22 +25,156 @@ def close; end
...
+              'severity': 'INFO',
+              'timestamp':'2019-01-08 14:51:39.701-0800',
+              'message':'hello world'
+            }
+          ]]
+        end
+      end
+
+      context 'when the log level is too low' do
+        it 'logs a thing' do
```
Comment (jobertabma) [inline spec/integration/fluent_logger_rails/logger_spec.rb:174]: doesn't log a thing

File: spec/integration/fluent_logger_rails/tagged_hash_formatter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    before do
+      formatter.parent_key = :payload
+    end
+
+    it 'formats the date' do
+      expect(formatter.call(0, DateTime.now, nil, 'hi')).to eq(
+        payload: {
+          severity: 'DEBUG',
+          message: 'hi',
+          timestamp: '2019-01-08T14:51:39.700999 '
```
Comment (jobertabma) [inline spec/integration/fluent_logger_rails/tagged_hash_formatter_spec.rb:31]: Where does this trailing whitespace come from? Is it really necessary? 🤔

Comment (bwillis) [reply spec/integration/fluent_logger_rails/tagged_hash_formatter_spec.rb:31]: Oh yeah, I noticed that before, good question. It looks like it's coming from the default `Logger::Formatter#format_datetime` function:
Screen Shot 2020-08-18 at 9 07 01 AM

It looks like it's probably being "helpful" and putting a space in there so you get separation between the date and the PID. `I, [2019-01-08T14:51:39.700999 #1234] ....`. Little odd for sure, I can add a default format that works better with a hash/json payload aka no trailing space.

File: spec/integration/fluent_logger_rails/logger_spec.rb (hunk 1 above)
Comment (bwillis) [reply spec/integration/fluent_logger_rails/logger_spec.rb:174]: Whoops, thanks!

**Pull request merged**